core/vm/instructions.go

```diff
@@ -585,6 +585,9 @@ func opJump(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byt
 	}
 	pos := scope.Stack.pop()
 	if !scope.Contract.validJumpdest(&pos) {
+		if !scope.Contract.UseGas(interpreter.evm.TxContext.Accesses.TouchCodeChunksRangeAndChargeGas(scope.Contract.CodeAddr[:], pos.Uint64(), 1, uint64(len(scope.Contract.Code)), false)) {
```

Comment:
Another option I thought was in `interpreter.go` do a:
```
if err == ErrInvalidJump {
  ...
}
```
since a single if would cover both JUMP(I) scenarios.

But we also need stack values, and that seems quite odd for that place in the code.

Note that this code addition is in instructions that won't be allowed by EOF, so this will work correctly in post-EOF world.

Reply:
shouldn't the location be touched regardless of its validity?

Reply:
If is valid, it will be touched by the [main interpreter loop](https://github.com/gballet/go-ethereum/blob/kaustinen-with-shapella/core/vm/interpreter.go#L182-L189).

Reply:
my bad I missed the `!`

Reply:
Explained offband. We add it to the witness if the jump is invalid.